actors/builtin/miner/miner_actor.go

```diff
@@ -1351,17 +1352,54 @@ func (a Actor) CompactPartitions(rt Runtime, params *CompactPartitionsParams) *a
 		rt.Abortf(exitcode.ErrIllegalArgument, "invalid deadline %v", params.Deadline)
 	}
 
+	store := adt.AsStore(rt)
 	var st State
 	rt.State().Transaction(&st, func() interface{} {
 		info := getMinerInfo(rt, &st)
 		rt.ValidateImmediateCallerIs(info.Worker)
 
+		// TODO: enforce this for the particular deadline only?
```

Comment:
Describe why this restriction, "... so we can calculate the termination fee"

Reply:
Actually, I have a question here: Can we narrow this restriction to partitions with pending early terminations? Currently, we prevent _all_ compactions if the miner has _any_ pending early terminations.

Reply:
Probably. But limiting to a deadline gets a long way, so I'm not sure it's worth it. They should really just process it.